lib/HLSL/DxilPrecisePropagatePass.cpp

```diff
@@ -128,30 +227,19 @@ static void PropagatePreciseAttributeOnPointer(
 
 static void
 PropagatePreciseAttribute(Instruction *I, DxilTypeSystem &typeSys,
-                          std::unordered_set<Instruction *> &processedSet) {
+                          ValueSet &processedSet) {
   LLVMContext &Context = I->getContext();
-  if (AllocaInst *AI = dyn_cast<AllocaInst>(I)) {
-    PropagatePreciseAttributeOnPointer(AI, typeSys, Context, processedSet);
-  } else if (dyn_cast<CallInst>(I)) {
-    // Propagate every argument.
-    // TODO: only propagate precise argument.
-    for (Value *src : I->operands())
-      PropagatePreciseAttributeOnOperand(src, typeSys, Context, processedSet);
-  } else if (dyn_cast<FPMathOperator>(I)) {
-    // TODO: only propagate precise argument.
-    for (Value *src : I->operands())
-      PropagatePreciseAttributeOnOperand(src, typeSys, Context, processedSet);
-  } else if (LoadInst *ldInst = dyn_cast<LoadInst>(I)) {
-    Value *Ptr = ldInst->getPointerOperand();
-    PropagatePreciseAttributeOnPointer(Ptr, typeSys, Context, processedSet);
-  } else if (GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(I))
-    PropagatePreciseAttributeOnPointer(GEP, typeSys, Context, processedSet);
-  // TODO: support more case which need
+  for (Value *src : I->operands())
+    PropagatePreciseAttributeOnOperand(src, typeSys, Context, processedSet);
+
+  if (PHINode *Phi = dyn_cast<PHINode>(I)) {
+    // TODO: For phi, we have to mark branch conditions that impact incoming paths.
```

Comment:
I think it should be fairly easy. Use control dependence to find the blocks that the phi preds are control dependent on and then propagate on the branches of those blocks.

Reply:
Thanks, we should add that in a subsequent PR.